packages/apollo-server-codegen/src/IR/Resolvers.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  FieldDefinitionNode,
+  FieldNode,
+  InputValueDefinitionNode,
+  OperationDefinitionNode,
+  parse,
+  StringValueNode
+} from "graphql";
+import { Translatable, Translator } from "../Translators";
+import { Description } from "./Descriptions";
+import { TypelessObjectDefinition } from "./Objects";
+import {
+  CompoundType,
+  findRootType,
+  makeType,
+  TypeDefinition,
+  NonNullType
+} from "./Types";
+
+const parseSelections = (source: string) =>
```

Comment:
Let's not copy and paste this around, but instead make it a common utility in the `apollo-graphql` package https://github.com/apollographql/apollo-server/blob/master/packages/apollo-gateway/src/utilities/graphql.ts#L100

Reply:
Would that mean adding it to apollo-graphql and concurrently updating the dependency in apollo-server and using it from there too? Or just add it to apollo-graphql for now and worry about apollo-server later?

Reply:
The latter is fine, concurrent isn't important - let's just add a note somewhere via a WIP PR or a JIRA ticket to make the switch. This would also make for a nice, small PR of its own that we could land immediately if you'd like.

Reply:
Learna is absolutely refusing to make "apollo-graphql" available within "apollo-server-codegen". I've tried everything, including manually writing the package-lock, and it just won't install it. Any tips are appreciated. @abernix in particular.

Reply:
Resolved.